tests/e2e_test.go

```diff
@@ -0,0 +1,293 @@
+			c.Equal(allRelays.From, test.date)
+			c.Equal(allRelays.To, test.date.AddDate(0, 0, 1))
+
+			/* /relays/apps */
+			allAppsRelays, err := get[[]api.AppRelaysResponse](relayMeterBaseURL, "v0/relays/apps", "", dateParams)
+			c.Equal(test.err, err)
+			for _, appRelays := range allAppsRelays {
+				c.Len(appRelays.Application, 64)
+				c.NotEmpty(appRelays.Count.Success)
+				c.NotEmpty(appRelays.Count.Failure)
+				c.Equal(appRelays.From, test.date)
+				c.Equal(appRelays.To, test.date.AddDate(0, 0, 1))
+			}
+
+			/* /relays/apps/{APP_PUB_KEY} */
+			appRelays, err := get[api.AppRelaysResponse](relayMeterBaseURL, "v0/relays/apps", test.appPubKey, dateParams)
+			c.Equal(test.err, err)
+			c.Len(appRelays.Application, 64)
+			c.NotEmpty(appRelays.Count.Success)
+			c.NotEmpty(appRelays.Count.Failure)
+			c.Equal(appRelays.From, test.date)
+			c.Equal(appRelays.To, test.date.AddDate(0, 0, 1))
+
+			/* /relays/users/{USER_ID} */
+			userRelays, err := get[api.UserRelaysResponse](relayMeterBaseURL, "v0/relays/users", test.userID, dateParams)
+			c.Equal(test.err, err)
+			c.Len(userRelays.User, 24)
+			c.Len(userRelays.Applications, 10)
+			c.Len(userRelays.Applications[0], 64)
+			c.NotEmpty(userRelays.Count.Success)
+			c.NotEmpty(userRelays.Count.Failure)
+			c.Equal(userRelays.From, test.date)
+			c.Equal(userRelays.To, test.date.AddDate(0, 0, 1))
+
+			/* /relays/endpoints */
+			allEndpointsRelays, err := get[[]api.LoadBalancerRelaysResponse](relayMeterBaseURL, "v0/relays/endpoints", "", dateParams)
+			c.Equal(test.err, err)
+			for _, endpointRelays := range allEndpointsRelays {
+				c.Len(endpointRelays.Endpoint, 24)
+				c.Len(endpointRelays.Applications, 1)
+				c.Len(endpointRelays.Applications[0], 64)
+				c.NotEmpty(endpointRelays.Count.Success)
+				c.NotEmpty(endpointRelays.Count.Failure)
+				c.Equal(endpointRelays.From, test.date)
+				c.Equal(endpointRelays.To, test.date.AddDate(0, 0, 1))
+			}
+
+			/* /relays/endpoints/{ENDPOINT_ID} */
+			endpointRelays, err := get[api.LoadBalancerRelaysResponse](relayMeterBaseURL, "v0/relays/endpoints", allEndpointsRelays[0].Endpoint, dateParams)
+			c.Equal(test.err, err)
+			c.Len(endpointRelays.Endpoint, 24)
+			c.Len(endpointRelays.Applications, 1)
+			c.Len(endpointRelays.Applications[0], 64)
+			c.NotEmpty(endpointRelays.Count.Success)
+			c.NotEmpty(endpointRelays.Count.Failure)
+			c.Equal(endpointRelays.From, test.date)
+			c.Equal(endpointRelays.To, test.date.AddDate(0, 0, 1))
+
+			/* /relays/origin-classification */
+			allOriginRelays, err := get[[]api.OriginClassificationsResponse](relayMeterBaseURL, "v0/relays/origin-classification", "", dateParams)
+			c.Equal(test.err, err)
+			for _, originRelays := range allOriginRelays {
+				c.Len(originRelays.Origin, 20)
+				c.NotEmpty(originRelays.Count.Success)
+				c.Equal(originRelays.From, test.date)
+				c.Equal(originRelays.To, test.date.AddDate(0, 0, 1))
+			}
+
+			/* /relays/origin-classification/{ORIGIN} */
```

Comment:
Please see above comment regarding breaking this into multiple test functions: each E2E test function will then focus on a specific aspect of the relay-meter.

Reply:
Same as above.

Reply:
Done, I've refactored these tests into their own funcs